### PR TITLE
Adding Bisection and Brent-Dekker

### PR DESCRIPTION
**Pull request by jkozdon**

### CODE REVIEW

File: src/RootSolvers.jl [hunk 1]
```diff
@@ -328,4 +354,233 @@ function find_zero(
...
+            x0, y0 = x2, y2
+        end
+        push_history!(x_history, x2, soltype)
+        push_history!(y_history, y2, soltype)
+    end
+    x = (x0 + x1) / 2
+    return SolutionResults(
+        soltype,
+        (x0 + x1) / 2,
+        true,
```
Comment (charleskawczynski) [inline src/RootSolvers.jl:395]: Shouldn't we have convergence check inside the iteration? Is this method guaranteed to converge after `maxiter`? Also, what if `maxiter` is passed and is not equal to the default?

Comment (jkozdon) [reply src/RootSolvers.jl:395]: Bisection is guaranteed to converge to a given bracket size in a given number of iterations. I'll update all this with your new functor stuff soon.

Comment (jkozdon) [reply src/RootSolvers.jl:395]: But you are right, that if maxiter is not the default there is no guaranteed convergence

**Comment by charleskawczynski** [comment]
Looking good! Also, this is making me think that maybe we can standardize an API. Each method could have properties for the things it needs. For example

```julia
struct SecantMethod{FT<:AbstractFloat} <: RootSolvingMethod
x0::FT
x1::FT
end
```
and
```julia
struct NewtonsMethod{F<:Function, FT<:AbstractFloat} <: RootSolvingMethod
f′::F
x0::FT
end
```

Then all methods can have a standardized interface (`find_zero(::f, ::RootSolvingMethod, ::SolutionType, ::AbstractTolerance, maxiters)`). Need not do this now, but what do you think?

**Comment by jkozdon** [comment]
> Also, this is making me think that maybe we can standardize an API.

I had thought of this as well.

**Comment by charleskawczynski** [comment]
I tried merging this with master [here](https://github.com/CliMA/RootSolvers.jl/compare/ck/jek/bracketed_merge_master), but I wasn't able to get the Bisection method working, and I didn't know off-hand how to generalize for the loop unrolling option. But feel free to pick up from here if you'd like.

**Comment by jkozdon** [comment]
> I tried merging this with master [here](https://github.com/CliMA/RootSolvers.jl/compare/ck/jek/bracketed_merge_master), but I wasn't able to get the Bisection method working, and I didn't know off-hand how to generalize for the loop unrolling option. But feel free to pick up from here if you'd like.

I think that we can just close it for now.

**Pull request closed without merging**